x-pack/plugins/uptime/server/lib/requests/search/enrich_monitor_groups.ts

```diff
@@ -311,25 +316,23 @@ const getHistogramForMonitors = async (
         },
       },
       aggs: {
-        by_id: {
-          terms: {
-            field: 'monitor.id',
-            size: queryContext.size,
+        histogram: {
+          auto_date_histogram: {
+            field: '@timestamp',
+            // 12 seems to be a good size for performance given
+            // long monitor lists of up to 100 on the overview page
+            buckets: 12,
```

Comment:
i think i agree with 12, though it can be made bit more adaptive, with something like 
queryContext.size < 50 then 25, WDYT?

Reply:
Perhaps, but maybe we can save that for a future PR?

Reply:
me and @justinkambic  discussed this and we decided to keep it at 12, since it's provides more uniform experience as well as even for UI, it will be more optimize to render in the table.